### PR TITLE
MGMT-17120: Delete device's associated EnrollmentRequest

### PR DESCRIPTION
**Pull request by oglok**
If Device resource is deleted, and its associated EnrollmentRequest persists, a re-enrollment process of the device won't be possible. This makes sure that enrollment requests are deleted with its device.

### CODE REVIEW

File: internal/service/device.go
Comment (mangelajo) [inline internal/service/device.go:125]: I would try to wrap the device delete and enrollment request delete in a transaction if we can.  Then a device cannot be deleted, something go wrong in the middle and the enrollment request left around.

**Comment by oglok** [comment]
let me know what you think now @mangelajo @hexfusion

**Review by hexfusion — COMMENTED** [review]
Seems reasonable for a cascading like delete defer to @avishayt or @mangelajo for the tag

**Comment by mangelajo** [comment]
> Seems reasonable for a cascading like delete defer to @avishayt or @mangelajo for the tag

What do you mean?

**Comment by hexfusion** [comment]
I just looked at the code briefly but I am not the authority for store. So wanted to make sure you or @avishayt didn't have opinions before it merged

**Comment by oglok** [comment]
I forgot to sign the commit. On my way...

File: internal/store/device.go
Comment (oglok) [reply internal/store/device.go]: done!

File: internal/store/device.go
Comment (avishayt) [inline internal/store/device.go:259]:
```suggestion
		associatedRecord := model.EnrollmentRequest{Resource: model.Resource{OrgID: orgId, Name: name}}
```

File: internal/store/device.go
Comment (avishayt) [inline internal/store/device.go:252]: Can revert these lines (see next comment)

File: internal/store/device.go
Comment (avishayt) [inline internal/store/device.go:265]:
```suggestion
		err = innerTx.Unscoped().Delete(&associatedRecord) // Ignore the error if the EnrollmentRequest does not exist
		if err != nil {
			// log.Warn?
		}
```

File: internal/store/device.go
Comment (avishayt) [inline internal/store/device.go:251]:
```suggestion
log := log.WithReqIDFromCtx(ctx, s.log)
```

File: internal/store/device.go
Comment (avishayt) [inline internal/store/device.go:266]:
```suggestion
			log.Warningf("failed to delete associated enrollment request: %v", err)
```

File: internal/store/device.go
Comment (avishayt) [inline internal/store/device.go:19]:
```suggestion
```

**Review by avishayt — APPROVED** [review]

**Pull request merged**
